pkg/ingester/circuitbreaker.go

```diff
@@ -211,17 +214,30 @@ func (cb *circuitBreaker) tryAcquirePermit() (bool, error) {
 // that lasted longer than the configured timeout are treated as a failure.
 // The returned error is only used for testing purposes.
 func (cb *circuitBreaker) finishPushRequest(duration time.Duration, pushErr error) error {
+	return cb.finishRequest(duration, cb.cfg.PushTimeout, pushErr)
+}
+
+// finishReadRequest should be called to complete the read request executed upon a
+// successfully acquired circuit breaker permit.
+// It records the result of the push request with the circuit breaker. Read requests
+// that lasted longer than the configured timeout are treated as a failure.
+// The returned error is only used for testing purposes.
+func (cb *circuitBreaker) finishReadRequest(readDuration time.Duration, readErr error) error {
+	return cb.finishRequest(readDuration, cb.cfg.ReadTimeout, readErr)
+}
+
+func (cb *circuitBreaker) finishRequest(actualDuration time.Duration, maximalDuration time.Duration, err error) error {
```

Comment:
```suggestion
func (cb *circuitBreaker) finishRequest(actualDuration time.Duration, maxDuration time.Duration, err error) error {
```

(Maybe or maybe `maxiumumDuration`)?

Reply:
I have changed it to `maximumAllowedDuration` that should be even more explanatory.